### PR TITLE
feat: show legend on supblots if a name is provided

### PR DESCRIPTION
**Pull request by MerlinSmiles**
Fixes part of #477.

Changes proposed in this pull request:
- when name is provided during `QtPlot.add()` a legend is added to that subplot.

Not going into more of #477 now.

@core @AdriaanRol

### CODE REVIEW

**Comment by MerlinSmiles** [comment]
haha, dont merge there is a mistake still...

**Comment by MerlinSmiles** [comment]
works now...

**Comment by AdriaanRol** [comment]
Looking cool 👍 . Any chance of a teaser picture?

**Comment by MerlinSmiles** [comment]
not much to show, looks like a qt plot with legend...

File: qcodes/plots/pyqtgraph.py [hunk 1]
```diff
@@ -99,6 +99,10 @@ def add_to_plot(self, subplot=1, **kwargs):
                 self.subplots.append(self.add_subplot())
         subplot_object = self.subplots[subplot - 1]
 
+        if 'name' in kwargs:
+            if subplot_object.legend is None:
+                subplot_object.addLegend(offset=(-30,30))
```
Comment (AdriaanRol) [inline qcodes/plots/pyqtgraph.py]: I assume this is in percent with respect to the total figure size?

Comment (MerlinSmiles) [inline qcodes/plots/pyqtgraph.py]: actually I think it is in (screen) pixels... I just moved it from top left to top right.

**Comment by AdriaanRol** [comment]
@giulioungaretti I'd be interested in seeing this pr of @MerlinSmiles merged but haven't seen any activity here, did anyone take a look?

**Comment by giulioungaretti** [comment]
@MerlinSmiles @MerlinSmiles Lgtm, by default name is not in the keawegs rigth ?

**Comment by MerlinSmiles** [comment]
@giulioungaretti i have not seen that anywhere. `name` is pyqtgrah default, while matplotlib uses `label`.

**Comment by giulioungaretti** [comment]
Cool, then we can merge and wait to see if somebody complains !

**Pull request merged**